expr/shaper.go

```diff
@@ -0,0 +1,418 @@
+package expr
+
+import (
+	"fmt"
+
+	"github.com/brimsec/zq/zcode"
+	"github.com/brimsec/zq/zng"
+	"github.com/brimsec/zq/zng/resolver"
+	"github.com/brimsec/zq/zson"
+)
+
+// A ShaperTransform represents one of the different transforms that a
+// shaper can apply.
+type ShaperTransform int
+
+const (
+	Cast = ShaperTransform(iota)
```

Comment:
NIt: Use conventional form.
```suggestion
	Cast ShaperTransform = iota
```